mingw_files/vcenv.sh

```diff
@@ -1,11 +0,0 @@
-export UCRTContentRoot='C:\Program Files (x86)\Windows Kits\10\'
-export INCLUDE='C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include;C:\Program Files (x86)\Windows Kits\10\Include\10.0.10240.0\ucrt;'
-export VCTargetsPath='C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\v140\'
-export Platform='X64'
-export VCINSTALLDIR='C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\'
-export LIB='C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\lib\amd64;C:\Program Files (x86)\Windows Kits\10\lib\10.0.10240.0\ucrt\x64;'
-export LIBPATH='C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\lib\amd64;'
-export WindowsSdkDir='C:\Program Files (x86)\Windows Kits\10\'
-export UCRTVersion='10.0.10240.0'
-export UniversalCRTSdkDir='C:\Program Files (x86)\Windows Kits\10'
-export PATH='/c/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/amd64:/c/Program Files (x86)/Windows Kits/10/bin/x64:/c/Program Files (x86)/Windows Kits/10/bin/x86':$PATH
```

Comment:
It is not directly related to one setup.py, but it opened up possibility of easily dropping this dependency. already working in https://github.com/neuronsimulator/nrn/pull/1339 (but not with `-j 2`, which is not an issue anymore with this PR)

Reply:
@alexsavulescu : I couldn't find the "exact" post that I wanted to highlight from #319 but some relevant threads about why compiling python extensions with MSVC is required:
* has some good references: https://matthew-brett.github.io/pydagogue/mingw_python.html
* not exact topic but I remember hitting this in #319: https://devblogs.microsoft.com/python/unable-to-find-vcvarsall-bat/
* (very old) discussion on python.org: https://bugs.python.org/issue23246

In summary, from my old memory what I remember is - you can build things with mingw and it might be working on your machine/python but there is no guarantee that it will on other machines reliably.

So I think we should really verify this item as I remember spending quite some time on this 2-3 years ago.

Reply:
Stands to reason. Maybe @ramcdougal  or @nrnhines can test the installer on their machines? I'm using a VM.
Link to installer: https://dev.azure.com/neuronsimulator/_apis/resources/Containers/26023319/drop?itemPath=drop%2Fnrn-nightly-AMD64.exe

Reply:
Just as a side note, Python has evolved and much of the older issues would eventually be fixed by now.

Reply:
> Just as a side note, Python has evolved and much of the older issues would eventually be fixed by now

This is true in general but here we are looking at a fundamental issue (of ABI) between mingw and msvc toolchains.  So unless somewhere this is explicitly stated the issue is solved, we shouldn't assume this is fixed. Based on my past experience, testing with 1-2 python won't give us a guarantee (if we get new python installer built with slightly different MSVC, this could cause segfaults). 

**_Until we find a solution / answer_**, to finalize this PR, would it be hard to keep vcenv.sh?

Reply:
Additionally, I am trying to remind myself bit more: some more references in: https://wiki.python.org/moin/WindowsCompilers

From https://github.com/cython/cython/issues/4600 Jan 22):

```console
The C compiler is handled by distutils and setuptools, not by Cython
MSVC is the only supported compiler to use with Python on Windows
It is just about possible to get mingw to work (https://stackoverflow.com/questions/70654722/how-to-create-c-extension-embed-python-with-mingw-w64-on-windows) but it is unsupported so if you run into bugs then you may not get any help
While mingw-w64-compiler is not really supported (the only supported windows compiler is [MSVC](https://wiki.python.org/moin/WindowsCompilers#Which_Microsoft_Visual_C.2B-.2B-_compiler_to_use_with_a_specific_Python_version_.3F)), it can be used to create C-extensions or to embed Python. There are however no guarantee, this won't break in the future versions.
```
Maybe we can do one experiment together tomorrow - with the presence and absence of vcenv.sh, we can check what `setuptools/_distutils/cygwinccompiler.py` ends up using.

Reply:
> would it be hard to keep vcenv.sh?

not entirely sure, doable

Reply:
xref: https://github.com/neuronsimulator/nrn/issues/319#issuecomment-554750099 
https://github.com/neuronsimulator/nrn/issues/319#issuecomment-555032976

Reply:
https://www.msys2.org/docs/python/

Reply:
Interesting : https://github.com/cython/cython/issues/4470#issuecomment-1361506709